### DISCUSSION TITLE
Run ioSender without connecting.

### DISCUSSION THREAD

**Discussion by phil-barrett** [terjeio/ioSender#222] [Ideas]
Currently ioSender will not load if it can not connect to a real Grbl board.  This gets reported as a bug fairly often.  Most other GCode Senders load first and then connect at a later point so it is not surprising that a lot of people expect the same behavior in ioSender.

I know that it is a fair amount of work but I think it is worth having a discussion about the pros and cons of changing ioSender to delay connecting until loaded and running.  I would like to hear what others have to say about this.

In favor of not changing the behavior:

- unnecessary because most people only use it with an actual Grbl controller.
- avoids an extra step of connecting (or additional work to do an automatic connection)
- the effort to do so takes away from other feature additions.

In favor of adding the change

- prevents confusion on the part of new users
- allows switching connections (or even different boards).  
- makes setting up ethernet a little easier (first have to find the ip addr via usb or serial)

On the ethernet set up point.  Here is how I tell people to do it now:

1. Start ioSender via USB.  Note the IP address in the console `[IP:xxx.xxx.xxx.xxx]`
2. Close ioSender
3. Unplug the USB cable from the grblHAL controller.  Plug the Ethernet Cable into the grblHAL controller.
4. Start ioSender. It may take a while for ioSender to time out on the USB connection.
5. At this point, it will put up the Sender connection dialog box that allows you to choose connection method.  Select Network and enter the above noted IP address. Do not click OK yet.
6. If your grblHAL board is powered via USB, reconnect the USB cable.
7. Click OK. ioSender should now come up connected via Ethernet.

With the above discussed modification (and the ability to disconnect, too), that could collapse into 4 steps with no unplugging/plugging of cables and no need to wait for the USB connection to time out.

**Comment by cheechNC** [comment]
Absolutely. I need to run the program in demo mode. There's not one notice anywhere I could find that the program doesn't run without a board connected. At least a pop up dialog with a sorry but..

**Reply by terjeio** [reply]
I have added info about how to use the grblHAL simulator to run the sender in a pretty realistic demo mode to the [readme](https://github.com/terjeio/ioSender/blob/master/readme.md). At some point I may allow starting the sender without a connection, but this will not provide the demo capabilities as the simulator offers.

**Reply by cheechNC** [reply]
thanks for the heads up @terjeio  Im excited about using grblHal and Phils ethernet b.o.b.s. I'll be checking it out carefully.
